src/glmfit.jl

```diff
@@ -52,35 +56,76 @@ end
 
 Update the mean, working weights and working residuals, in `r` given a value of
 the linear predictor, `linPr`.
-
-This the fallback method.  Specialized methods are defined for various
-distributions and their canonical link functions.
 """
 function updateμ!{T<:FPVector,D,L}(r::GlmResp{T,D,L}, linPr::T)
-    y, η, μ, μη, wts, wrkwt, wrkres, devres =
-        r.y, r.eta, r.mu, r.mueta, r.wts, r.wrkwt, r.wrkresid, r.devresid
+    isempty(r.offset) ? copy!(r.eta, linPr) : broadcast!(+, r.eta, linPr, r.offset)
+    updateμ!(r)
+    if !isempty(r.wts)
+        r.devresid .*= r.wts
+        r.wrkwt .*= r.wts
+    end
+    r
+end
+
+function updateμ!{T<:FPVector,D<:Union{Bernoulli,Binomial}}(r::GlmResp{T,D,LogitLink})
+    y, η, μ, wrkres, wrkwt, dres = r.y, r.eta, r.mu, r.wrkresid, r.wrkwt, r.devresid
```

Comment:
Unrelated, but we could probably make `GlmResp` immutable, which would avoid the need to defining temporary variables to ensure performance.